source/run-command/run-command.rst

```diff
@@ -0,0 +1,185 @@
+===========
+Run Command
+===========
+
+:Title: Run Command
+:Status: Implementing
+:Minimum Server Version: N/A
+
+.. contents::
+
+--------
+
+Abstract
+========
+This specification defines requirements and behaviors for drivers' run command and related APIs.
+
+
+META
+====
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this
+document are to be interpreted as described in `RFC 2119
+<https://www.ietf.org/rfc/rfc2119.txt>`_.
+
+Specification
+=============
+
+-----
+Terms
+-----
+Command
+    A structure representing a BSON document that has a shape matching a supported MongoDB operation.
+
+---------------------------
+Implementation requirements
+---------------------------
+
+All drivers MAY offer the operations defined in the following sections.
+This does not preclude a driver from offering more.
+
+--------------
+``runCommand``
+--------------
+
+The following represents how a runCommand API SHOULD be exposed.
+
+.. code:: typescript
+
+    interface Db {
+      /**
+       * Takes an argument representing an arbitrary BSON document and executes it against the server.
+       */
+      runCommand(command: BSONDocument, options: RunCommandOptions): BSONDocument;
+    }
+
+    interface RunCommandOptions {
+      /**
+       * An optional readPreference setting to apply to server selection logic.
+       * This value MUST be applied to the command document as the $readPreference global command argument if not set to primary.
+       *
+       * @defaultValue primary
```

Comment:
```suggestion
       * @defaultValue ReadPreference(mode: primary)
```

This may be pedantic, but this is just the mode option. I think expressing the entire [ReadPreference structure](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#read-preference) would be more clear.

I don't recall offhand if the `$readPrefrence` [global command argument]9https://github.com/mongodb/specifications/blob/master/source/message/OP_MSG.rst#command-arguments-as-payload) takes all of the same options as the structure defined in the Server Selection spec, but the example does show it expressed as an object with a `mode` property, so I think that would still better match the suggested replacement here.

Reply:
The node driver, at least, will send mode, tags, maxStalenessSeconds, and hedge